queries/futures/useGetFuturesPositionForAccount.ts

```diff
@@ -1,54 +1,108 @@
 
-	return useQuery<PositionHistory[] | null>(
-		QUERY_KEYS.Futures.AccountPositions(
-			walletAddress,
-			network.id as NetworkId,
-			selectedAccountType
-		),
+	return useQuery<PositionHistoryState>(
+		QUERY_KEYS.Futures.AccountPositions(account ?? walletAddress, network.id as NetworkId),
 		async () => {
-			if (!walletAddress) return [];
+			if (!walletAddress) {
+				if (!account) setPositionHistory(DEFAULT_POSITION_HISTORY);
+				return DEFAULT_POSITION_HISTORY;
+			}
+
 			try {
-				const response = await request(
+				const response = await getFuturesPositions(
 					futuresEndpoint,
-					gql`
-						${FUTURES_POSITION_FRAGMENT}
-						query userAllPositions($account: String!, $accountType: String!) {
-							futuresPositions(where: { account: $account, accountType: $accountType }) {
-								...FuturesPositionFragment
-							}
-						}
-					`,
-					{ account: walletAddress, accountType: selectedAccountType }
+					{
+						where: {
+							account: account ?? walletAddress,
+						},
+					},
+					{
+						id: true,
+						lastTxHash: true,
+						openTimestamp: true,
+						closeTimestamp: true,
+						timestamp: true,
+						market: true,
+						asset: true,
+						account: true,
+						abstractAccount: true,
+						accountType: true,
+						isOpen: true,
+						isLiquidated: true,
+						trades: true,
+						totalVolume: true,
+						size: true,
+						initialMargin: true,
+						margin: true,
+						pnl: true,
+						feesPaid: true,
+						netFunding: true,
+						pnlWithFeesPaid: true,
+						netTransfers: true,
+						totalDeposits: true,
+						fundingIndex: true,
+						entryPrice: true,
+						avgEntryPrice: true,
+						lastPrice: true,
+						exitPrice: true,
+					}
 				);
-				return response?.futuresPositions ? mapTradeHistory(response.futuresPositions, true) : [];
+				const positionHistory = response ? mapFuturesPositions(response) : [];
+
+				const positionHistoryByType = positionHistory.reduce(
+					(acc: PositionHistoryState, position: PositionHistory) => {
+						const accountType = position.accountType;
+
+						// make sure it's not a duplicate before adding to the list
+						const existingPositionId = acc[accountType].findIndex((pos) => pos.id === position.id);
+						if (
+							existingPositionId === -1 ||
+							position.timestamp > acc[accountType][existingPositionId].timestamp
+						)
+							acc[accountType] = [
+								position,
+								...acc[accountType].filter(({ id }) => id !== existingPositionId),
+							];
+						return acc;
+					},
+					DEFAULT_POSITION_HISTORY
+				);
+
+				if (!account) setPositionHistory(positionHistoryByType);
+				return positionHistoryByType;
 			} catch (e) {
 				logError(e);
-				return null;
+				if (!account) setPositionHistory(DEFAULT_POSITION_HISTORY);
+				return DEFAULT_POSITION_HISTORY;
 			}
 		},
 		{
-			enabled: !!selectedFuturesAddress,
-			refetchInterval: 5000,
+			enabled: !!walletAddress,
```

Comment:
Having this condition means the state reset on line 32 won't be called as the query will be disabled when no wallet, I think we can just remove this check. I ended up doing this for a couple of other queries to ensure state clears when wallet disconnected.

Reply:
This will be fixed now with the "two query" setup. The one that sets state runs a `useEffect` if the query changes, and if the query data is not available it resets to the default with no positions. This allows us to pass the `walletAddress` parameter which might be null, but know that nothing will run and the state will be reset if this happens.